test/parser-test.js

```diff
@@ -1,116 +1,188 @@
 // LICENSE : MIT
 "use strict";
 import assert from "power-assert"
-import {parse, containIncludeCommand, splitLabelToCommands} from "../src/parser"
+import {parse, parseVariablesFromLabel, containIncludeCommand, splitLabelToCommands} from "../src/parser"
 var content = `
-[include](fixtures/test.js)
+[include,title:"test.js"](fixtures/test.js)
 `;
 describe("parse", function () {
     context("translate lang", function () {
         it("should return object for replace", function () {
             var results = parse(content, __dirname);
             assert(results.length > 0);
             var result = results[0];
-            assert.equal(result.target, "[include](fixtures/test.js)");
-            var expected = '> <a name="test.js" href="fixtures/test.js">test.js</a>\n'
+            assert.equal(result.target, "[include,title:\"test.js\"](fixtures/test.js)");
+//------------------------------------------------------------------------------
+            var expected =`\
+        {% if file.type=="asciidoc" %}
+> .link:fixtures/test.js[Code 0: test.js]
+anchor:test.js[Code 0]
+        {% else %}
+> <a id="test.js" href="fixtures/test.js">Code 0: test.js</a>
+        {% endif %}
+`
                 + '\n'
                 + '``` javascript\nconsole.log(\"test\");\n```';
+//------------------------------------------------------------------------------
             assert.equal(result.replaced, expected);
         });
         it("should prefer use lang-<aceMode>", function () {
-            var exs_content = `[include, lang-typescript](fixtures/test.ts)`;
+            var exs_content = `[include, title:"test.ts", lang-typescript](fixtures/test.ts)`;
             var results = parse(exs_content, __dirname);
             assert(results.length > 0);
             var result = results[0];
-            assert.equal(result.target, "[include, lang-typescript](fixtures/test.ts)");
-            var expected = '> <a name="test.ts" href="fixtures/test.ts">test.ts</a>\n'
+            assert.equal(result.target, "[include, title:\"test.ts\", lang-typescript](fixtures/test.ts)");
```

Comment:
if you want to avoid writing escaping `\"`, use `'` single quote instead of `"`.

```
'[include, title:"test.ts", lang-typescript](fixtures/test.ts)'
````

is euqal to

```
"[include, title:\"test.ts\", lang-typescript](fixtures/test.ts)"
```

in JavaScript.